src/components/legend.ts

```diff
@@ -300,23 +301,23 @@ export module Component {
     }
 
     /**
-     * Gets the SymbolGenerator of the legend, which dictates how
+     * Gets the SymbolGeneratorAccessor of the legend, which dictates how
      * the symbol in each entry is drawn.
      *
-     * @returns {SymbolGenerator} The SymbolGenerator of the legend
+     * @returns {symbolGeneratorAccessor} The SymbolGeneratorAccessor of the legend
      */
-    public symbolGenerator(): SymbolGenerator;
+    public symbolGeneratorAccessor(): SymbolGeneratorAccessor;
     /**
-     * Sets the SymbolGenerator of the legend
+     * Sets the SymbolGeneratorAccessor of the legend
      *
      * @returns {Legend} The calling Legend
      */
-    public symbolGenerator(symbolGenerator: SymbolGenerator): Legend;
-    public symbolGenerator(symbolGenerator?: SymbolGenerator): any {
-      if (symbolGenerator == null) {
-        return this._symbolGenerator;
+    public symbolGeneratorAccessor(symbolGeneratorAccessor: SymbolGeneratorAccessor): Legend;
+    public symbolGeneratorAccessor(symbolGeneratorAccessor?: SymbolGeneratorAccessor): any {
```

Comment:
You still have to look up what `SymbolGeneratorAccessor` is though, so that's not really helpful.